openedx_tagging/core/tagging/import_export/parsers.py

```diff
@@ -0,0 +1,331 @@
+"""
+Parsers to import and export tags
+"""
+import csv
+import json
+from enum import Enum
+from io import BytesIO, TextIOWrapper, StringIO
+from typing import List, Tuple
+
+from django.utils.translation import gettext_lazy as _
+
+from .import_plan import TagItem
+from .exceptions import (
+    TagParserError,
+    InvalidFormat,
+    FieldJSONError,
+    EmptyJSONField,
+    EmptyCSVField,
+)
+from ..models import Taxonomy
+from ..api import get_tags
+
+
+class ParserFormat(Enum):
+    """
+    Format of import tags to taxonomies
+    """
+
+    JSON = ".json"
+    CSV = ".csv"
+
+
+class Parser:
+    """
+    Base class to create a parser
+
+    This contains the base functions to load data,
+    validate required fields and convert tags to DLS format
```

Comment:
This contains the base functions to convert between
a simple file format like CSV/JSON and a list of TagItems.
It can convert in both directions, for use during import or export.

Reply:
Updated: 66b17c9a621f2f9cc0e72e087f4c6cc03696d365